### PR TITLE
MDCT-1860: Implement Auth Timeout

### PR DESCRIPTION
**Pull request by braxex**
## Code author checklist
- [ ] I have performed a self-review of my code
- [ ] I have added [thorough](https://bit.ly/3zPrxuZ) tests
- [ ] I have added analytics, if necessary
- [ ] I have updated the documentation, if necessary

## Reviewer checklist (two different people)
- [ ] I have done the deep review and verified the items in the above checklist are g2g
- [ ] I have done the lgtm review

### CODE REVIEW

File: services/ui-auth/serverless.yml [hunk 1]
```diff
@@ -123,11 +123,19 @@ resources:
...
           - implicit
+          - code
         AllowedOAuthFlowsUserPoolClient: true
         AllowedOAuthScopes:
           - email
           - openid
           - profile
+        AccessTokenValidity: 2
+        IdTokenValidity: 2
+        RefreshTokenValidity: 5
```
Comment (gmrabian) [inline services/ui-auth/serverless.yml:134]: Just making sure we set these to desired values before we release
Id Token: "You can set the ID token expiration to any value between 5 minutes and 1 day."
Refresh token: "you can set the application's refresh token expiration to any value between 60 minutes and 10 years." (10 years 😰 )
Access token: "You can set the access token expiration to any value between 5 minutes and 1 day."
source: [Cognito](https://docs.aws.amazon.com/cognito/latest/developerguide/amazon-cognito-user-pools-using-the-access-token.html) and similar pages

File: services/ui-src/src/components/app/AppRoutes.test.tsx [hunk 2]
```diff
@@ -3,36 +3,43 @@ import { act } from "react-dom/test-utils";
...
+    window.HTMLElement.prototype.scrollIntoView = tempScroll;
+  });
   test("/admin is visible for admin user", async () => {
     const currentPath = history.location.pathname;
     expect(currentPath).toEqual("/admin");
   });
 });
 
-describe("Test AppRoutes for non-admin-specific routes", () => {
+describe("brax Test AppRoutes for non-admin-specific routes", () => {
```
Comment (BearHanded) [inline services/ui-src/src/components/app/AppRoutes.test.tsx:42]: I don't know that Brax needs to sign his tests

File: services/ui-src/src/constants.ts [hunk 3]
```diff
@@ -68,3 +68,8 @@ export enum States {
   WI = "Wisconsin",
   WY = "Wyoming",
 }
+
+// TIMEOUT PARAMS
+
+export const IDLE_WINDOW = 2 * 60 * 1000; // ms
+export const PROMPT_AT = 1 * 60 * 1000; //ms
```
Comment (BearHanded) [inline services/ui-src/src/constants.ts:75]: Correct the timeouts here

File: services/ui-auth/serverless.yml (hunk 1 above)
Comment (BearHanded) [reply services/ui-auth/serverless.yml:134]: these are the carts values:
```        AccessTokenValidity: 30
        IdTokenValidity: 30
        RefreshTokenValidity: 24
        TokenValidityUnits: 
          AccessToken: "minutes"
          IdToken: "minutes"
          RefreshToken: "hours"```

**Review by BearHanded — CHANGES_REQUESTED** [review]

**Review by BearHanded — APPROVED** [review]

File: services/ui-src/src/components/layout/Timeout.tsx [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+
+    if (timeLeft <= 0) {
+      logout();
+    }
+
+    const timer = setTimeout(() => {
+      setTimeLeft(calculateRemainingSeconds(expirationTime));
+    }, 1000);
+    return () => {
+      clearInterval(timer);
```
Comment (braxex) [inline services/ui-src/src/components/layout/Timeout.tsx:51]: Request: Should change to `clearTimeout` instead.

File: services/ui-auth/serverless.yml (hunk 1 above)
Comment (braxex) [reply services/ui-auth/serverless.yml]: 10 years sounds 👌

File: services/ui-src/src/components/app/AppRoutes.test.tsx (hunk 2 above)
Comment (braxex) [reply services/ui-src/src/components/app/AppRoutes.test.tsx]: lol sometimes i like to leave a calling card, ok?

**Review by britt-mo — APPROVED** [review]

**Pull request merged**